### ISSUE TITLE
After version upgrade to 6.5.2 we are facing issue in accessing dashboards which are using Cloudwatch as datasource

### ISSUE THREAD

**Issue by rspljignesh** [grafana/grafana#21411]
**What happened**: dashboard like Amazon EBS, Amazon EC2 stopped working and and giving error `Cloudwatch GetMetricData error: Maximum number of allowed metrics exceeded. Your search may have been limited.` I have already open support ticket to AWS for increasing the  Limit but they reply and informed to check with Grafana Developer. 

**What you expected to happen**: Grafana version upgrade from 5.4.2 to 6.5.2

**How to reproduce it (as minimally and precisely as possible)**: running Dashboard with datasource as cloudwatch with more then 100 Metrics in single request.

**Anything else we need to know?**: Here is reply of Amazon Support for reference
```GetMetricData API has a hard limit (cannot be increased) of 100 metrics per API call, so if Grafana dashboard is configured to fetch more than 100 metrics in a single call, this error was always pop-up.

It’s possible that Grafana is using SEARCH() math function to find the list of metrics and then doing GetMetricData for all the metrics in a single call.

You need to connect with Grafana to better understand how to overcome this limitation as the service team is not sure how Grafana is making use of GetMetricData API. Grafana should probably update their code to do multiple API calls if more than 100 metrics are needed to be fetched. ```

**Environment**:  Kubernetes Deployment
- Grafana version: 6.5.2
- Data source type & version: Cloudwatch
- OS Grafana is installed on: last week
- User OS & Browser: Chrome
- Grafana plugins: 
- Others:

**Comment by aocenas** [comment]
May be a duplicate of https://github.com/grafana/grafana/issues/20652 but not sure.

@sunker can you take a look at this?

**Comment by sunker** [comment]
@rspljignesh, in 6.5 we stop using the GetMetricStatistics api and use the GetMetricData api instead.  Please read the [what's new](https://grafana.com/docs/grafana/latest/guides/whats-new-in-v6-5/) and the [upgrading to 6.5 doc](https://grafana.com/docs/grafana/latest/installation/upgrading/#upgrading-to-v6-5). 

I suggest splitting the query into multiple panels.

**Comment by peycho** [comment]
Hey guys.

I'm not sure that I have to open a new issue, so I will just ask here.

I have the same issue with a single panel in my dashboard.

Grafana v7.4.0 (c2203b9859) using Ubuntu 16.04.7 LTS

Graph json configuration: https://gist.github.com/peycho/8d07956db897f1d1466a91d32a1bb516

![Screenshot from 2021-02-18 10-36-18](https://user-images.githubusercontent.com/170853/108328623-80ee3680-71d5-11eb-803c-c85ef32e0468.png)
![Screenshot from 2021-02-18 10-36-34](https://user-images.githubusercontent.com/170853/108328643-851a5400-71d5-11eb-8085-08a44fd83e6d.png)

Is there any configuration mistake of this is something that AWS should handle?

**Comment by sunker** [comment]
The API has a limit of 100 returned metrics, and that limit is reached in your query. The red warning is there to indicate that your query only received a partial response. I'd try to be more specific in the query so that you narrow down the search to fewer metrics.

**Comment by peycho** [comment]
I'm looking up only for one metric "CPUCreditUsage", but we have maybe 1000+ instances that are T class instances.
The problem is that they aren't static and are frequently replaced for scaling or other reasons.

**Comment by sunker** [comment]
Okay - didn't know it has been changed from 100 to 500. Then that is the limit that you've reached.

**Comment by peycho** [comment]
Can you give me some guidance how to avoid that or make it work properly?

**Comment by sunker** [comment]
I don't even now what kind of query it is that you have. But I guess you could play around with the time range and add/change the dimensions that you use. Also, refer to the [cloudwatch search expressions docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/search-expression-syntax.html) and our [community site](https://community.grafana.com/) for these kind of questions.

**Comment by peycho** [comment]
There is nothing complex in the query I'm using.

It's just a single metric but it comes from many instances.

```
[
   {
      "Expression":"REMOVE_EMPTY(SEARCH('{\"AWS/EC2\",\"InstanceId\"} MetricName=\"CPUCreditUsage\"', 'Average', 600))",
      "ID":"queryA",
      "Period":600
   }
]
```

![Screenshot from 2021-02-18 14-55-04](https://user-images.githubusercontent.com/170853/108360012-557d4300-71f9-11eb-94a5-02e9edb65f7a.png)

I've tried already with different time range. I can't really change the dimension since I need all the ec2 instances.

My real questions are;
Is this achievable at the current version?
I should fill a feature request, so Grafana can support all available metrics in the query?

**Comment by sunker** [comment]
> Is this achievable at the current version?
I can't answer that, but what I can say is that there's no way for grafana to overcome the 500 metrics per response limit. 
> I should fill a feature request, so Grafana can support all available metrics in the query?
You can forward that information to the CloudWatch team. Maybe they can increase the limit, I don't know.

**Comment by peycho** [comment]
Hi

I already spoke with the CloudWatch team. This is a hard limit and cannot be changed. https://github.com/grafana/grafana/issues/21411#issuecomment-781251828

> In order to avoid this error make sure to distribute the metrics in multiple GetMetricData call.

So Grafana doesn't support such multiple GetMetricData calls. That was my idea of the feature request.

Thanks for you time.
I appreciate it.

**Comment by viquer** [comment]
Hello,

have you considered using Metric Insights queries on Grafana through the CloudWatch connector? [Documentation](https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/aws-cloudwatch/query-editor/#create-a-metric-insights-query). 

Example of query similar to what you where looking for

```
SELECT SUM(CPUCreditUsage) FROM SCHEMA("AWS/EC2", InstanceId) GROUP BY InstanceId ORDER BY AVG() DESC LIMIT 10
```

This (^) query will return the top 10 Instances based on credit usage. You can also get the overall Credit Usage removing the `GROUP BY`:

```
SELECT SUM(CPUCreditUsage) FROM SCHEMA("AWS/EC2", InstanceId)
```

Note this has a limitation of only showing the latest 3h of metric, but can help unblock you.
